components/appmgt/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java

```diff
@@ -1437,6 +1435,68 @@ public static NativeArray jsFunction_getSubscribedAPIsByUsers(Context cx, Script
     }
 
     /**
+     * This method returns the endpoint for the webapps
+     * @param cx      Rhino context
+     * @param thisObj Scriptable object
+     * @param args    Passing arguments
+     * @param funObj  Function object
+     * @return Native array
+     * @throws org.wso2.carbon.appmgt.api.AppManagementException
+     */
+    public static NativeArray jsFunction_getAppsByEndPoint(Context cx, Scriptable thisObj,
+                                                                  Object[] args,
+                                                                  Function funObj)
+            throws AppManagementException {
+        NativeArray myn = new NativeArray(0);
+        APIProvider apiProvider = getAPIProvider(thisObj);
+
+        try {
+            /**
+             * http and https endpoint resolving by reading AppManagerConfiguration
+             */
+            AppManagerConfiguration config = HostObjectComponent.getAPIManagerConfiguration();
+            List<Environment> environments = config.getApiGatewayEnvironments();
+            String envDetails = "";
+            String httpEndpoint = null, httpsEndpoint = null, endpoint = null;
+            for(int i=0; i<environments.size(); i++){
+                Environment environment = environments.get(i);
+                envDetails = environment.getApiGatewayEndpoint();
+                try {
+                    httpEndpoint = envDetails.split(",")[0];
+                    httpsEndpoint = envDetails.split(",")[1];
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    handleException("Error occurred while getting endpoint form AppManagerConfiguration", e);
+                }
+            }
+
+            List<WebApp> apiList = apiProvider.getAppsWithEndpoint();
+            if (apiList != null) {
+                Iterator it = apiList.iterator();
+                int i = 0;
+                while (it.hasNext()) {
+                    NativeObject row = new NativeObject();
+                    Object appObject = it.next();
+                    WebApp app = (WebApp) appObject;
+                    APIIdentifier apiIdentifier = app.getId();
+                    row.put("name", row, apiIdentifier.getApiName());
+                    /**
+                     * this WebApp is for read the registry values

```

Comment:
Change the comment with correct formatting. Use // for line comments and Capitalize the first character.